### PR TITLE
Add C implementation for `@stdlib/math/base/special/acovercos`

### LINKED ISSUE #775: [RFC]: Add C implementation for `@stdlib/math/base/special/acovercos`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/acovercos`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acovercos).

```c
#include "stdlib/math/base/special/asin.h"

double stdlib_base_acovercos( const double x ) {
    return stdlib_base_asin( 1.0 + x );
}
```
Package: `@stdlib/math/base/special/acovercos`

### Related Issues

None.

### Questions

No.

### Other

Reference packages:
- [`@stdlib/math/base/special/acosh`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acosh)

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by adarshpalaskar1**
Resolves #775.

## Description

> This PR adds C implementation for [@stdlib/math/base/special/acovercos](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acovercos).

## Related Issues

> Does this pull request have any related issues?
None

This pull request:

-   resolves #775

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/acovercos/README.md [hunk 1]
```diff
@@ -105,6 +105,91 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/special/acovercos.h"
+```
+
+#### stdlib_base_acovercos( x )
+
+Computes the [inverse coversed cosine][inverse-coversed-cosine] of a double-precision floating-point number (in radians).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/README.md]:
```suggestion
Computes the [inverse coversed cosine][inverse-coversed-cosine] of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acovercos/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// MAIN //
+
+bench( pkg+'::native', opts, function benchmark( b ) {
+	var x;
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu()*-2.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/benchmark/benchmark.native.js]:
```suggestion
		x = ( -2.0*randu() ) + 0.0;
```

Keep the same as in the non-native.

File: lib/node_modules/@stdlib/math/base/special/acovercos/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#include <stdio.h>
+
+int main( void ) {
+	const double x[] = { -2.0, -1.80, -1.78, -1.67, -0.56, -0.27, -1.67, -0.78, -1.89, 0.0 };
+
+	double v;
+	int i;
+	for ( i = 0; i < 10; i++ ) {
+		v = stdlib_base_acovercos( x[ i ] );
+		printf( "acoversin(%lf) = %lf\n", x[ i ], v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/examples/c/example.c]:
```suggestion
		printf( "acovercos(%lf) = %lf\n", x[ i ], v );
```

File: lib/node_modules/@stdlib/math/base/special/acovercos/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Computes the inverse cotangent of a double-precision floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/lib/native.js]:
```suggestion
* Computes the inverse coversed cosine of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acovercos/lib/native.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+* var v = acovercos( 3.141592653589793/6.0 );
+* // returns NaN
+*
+* @example
+* var v = acovercos( -1.0 );
+* // returns 0.0
+*
+* @example
+* var v = acovercos( NaN );
+* // returns NaN
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/lib/native.js]: These are not correct. Copy and paste from the JS implementation: https://github.com/stdlib-js/stdlib/blob/615bcf165b0a9a9aad5f6591024e2dd1655c8914/lib/node_modules/%40stdlib/math/base/special/acovercos/lib/main.js. Meaning there is no need for these examples to differ from the JS variant.

File: lib/node_modules/@stdlib/math/base/special/acovercos/lib/native.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+
+// MAIN //
+
+/**
+* Computes the inverse coversed cosine of a double-precision floating-point number.
+*
+* @private
+* @param {number} x - input value
+* @returns {number} inverse coversed cosine (in radians)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/lib/native.js]:
```suggestion
* @returns {number} inverse coversed cosine
```

File: lib/node_modules/@stdlib/math/base/special/acovercos/manifest.json [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+          "./include"
+        ],
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/special/asin"
+        ]
+      }
+    ]
+  }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acovercos/manifest.json]:
```suggestion
  } 

```

**Review by kgryte — CHANGES_REQUESTED** [review]
@adarshpalaskar1 Thanks for working on this. The main outstanding thing which needs to be finished based on my initial review are the examples in the `/lib/native.js` file.

@Pranavchiku Given that you worked previously on similar packages, does this PR satisfy the necessary changes?

**Comment by Pranavchiku** [comment]
I'll check those once he applies code review and revert back.

File: lib/node_modules/@stdlib/math/base/special/acovercos/README.md [hunk 8]
```diff
@@ -105,6 +105,91 @@ for ( i = 0; i < x.length; i++ ) {
...
+```c
+#include "stdlib/math/base/special/acovercos.h"
+```
+
+#### stdlib_base_acovercos( x )
+
+Computes the [inverse coversed cosine][inverse-coversed-cosine] of a double-precision floating-point number.
+
+```c
+double out = stdlib_base_acovercos( 3.141592653589793/2.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acovercos/README.md]:
```suggestion
double out = stdlib_base_acovercos( -3.141592653589793/2.0 );
```

**Review by Pranavchiku — APPROVED** [review]
Looks good to me. Thanks @adarshpalaskar1 !

**Pull request merged**